CMakeLists.txt

```diff
@@ -0,0 +1,127 @@
+    set(${deps_var} ${ARGV} PARENT_SCOPE)
+endmacro()
+
+macro(add_leatherman_library)
+    add_library(${libname} STATIC ${ARGV})
+    set(${lib_var} "${libname}" PARENT_SCOPE)
+endmacro()
+
+macro(add_leatherman_test)
+    add_library(${testlibname} STATIC ${ARGV})
+    set(${testlib_var} "${testlibname}" PARENT_SCOPE)
+endmacro()
+
+# This macro gives us a pile of common variable names and behaviors
+# for leatherman libraries that we can work with
+#
+# * all the common LEATHERMAN_<LIBNAME>_{LIB,LIBS,INCLUDE,DEPS}
+#   variables for integrating individual leatherman libraries into a
+#   top-level project
+# * automatic test integration with the top-level test executable
+# * variables passed into each subdir for common naming conventions
+# * options to enable/disable individual libraries
+macro(add_leatherman_dir dir)
+    debug("Setting up leatherman library for ${dir}")
+    string(MAKE_C_IDENTIFIER "${dir}" id)
+    string(TOUPPER "${id}" id_upper)
+    set(option "LEATHERMAN_USE_${id_upper}")
+    set(include_dir "${CMAKE_CURRENT_SOURCE_DIR}/${dir}/inc")
+    set(libname "leatherman_${id}")
+    set(testlibname "${libname}_test")
+
+    if(DEFINED ${option})
+	debug("${option} is already set, using it")
+	set(enabled ${${option}})
+    else()
+	debug("${option} unset, using default")
+	set(enabled ${LEATHERMAN_DEFAULT_ENABLE})
+    endif()
+    option(${option} "Should ${dir} be built and used?" ${enabled})
+    if (${${option}})
+	set(include_var "LEATHERMAN_${id_upper}_INCLUDE")
+	set(lib_var "LEATHERMAN_${id_upper}_LIB")
+	set(deps_var "LEATHERMAN_${id_upper}_DEPS")
+	set(testlib_var "LEATHERMAN_${id_upper}_TESTLIB")
+ 
+	set(${include_var} ${include_dir})
+	set(${lib_var} "") # if library is built, this will be set automatically
+	set(${testlib_var} "") # if tests are specified, this will be set automatically
+
+	# By adding the subdirectory after setting all variables, but
+	# before exporting, we give the library an opportunity to
+	# munge them (for example, to add vendor dirs)
+	add_subdirectory("${dir}")
+
+	# We set this one afterwards because it doesn't need
+	# overriding
+	set(libs_var "LEATHERMAN_${id_upper}_LIBS")
+	set(${libs_var} ${${lib_var}} ${${deps_var}})
+
+	export_var(${include_var})
+	export_var(${lib_var})
+	export_var(${libs_var})
+	export_var(${deps_var})
+
+	if(NOT "${ARGV1}" STREQUAL EXCLUDE_FROM_VARS)
+	    debug("Appending values for ${id_upper} to common vars")
+	    list(APPEND LEATHERMAN_INCLUDE_DIRS ${${include_var}})
+	    list(APPEND LEATHERMAN_LIBRARIES ${${libs_var}})
+	    list(APPEND LEATHERMAN_TESTLIBS ${${testlib_var}})
+	else()
+	    debug("Excluding values for ${id_upper} from common vars")
+	endif()
+    endif()
+endmacro(add_leatherman_dir)
+
+add_leatherman_dir(catch EXCLUDE_FROM_VARS)
+add_leatherman_dir(locale)
+
+export_var(LEATHERMAN_INCLUDE_DIRS)
+export_var(LEATHERMAN_LIBRARIES)
+
+if(LEATHERMAN_ENABLE_TESTING)
+    enable_testing()
+    add_subdirectory(tests)
```

Comment:
That's the intended pattern. This is just for the test executable. I'll rename the directory and clarify that with a comment down here.